src/components/ui/drawers/NameEditionDrawer.tsx

```diff
@@ -0,0 +1,176 @@
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerDescription,
+  DrawerFooter,
+  DrawerTitle,
+  DrawerTrigger,
+} from '@/components/ui/drawer';
+import { Input } from '@/components/ui/input';
+import { Button } from '@/components/ui/button';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { FormMessage } from '@/components/ui/formMessage';
+import {
+  shareHouseNameSchema,
+  TShareHouseNameSchema,
+  categoryNameSchema,
+  TCategoryNameSchema,
+} from '@/constants/schema';
+import { INPUT_TEXTS } from '@/constants/input-texts';
+
+const { SHAREHOUSE_NAME, CATEGORY_NAME } = INPUT_TEXTS;
+
+interface INameEditionDrawer {
+  name: string;
+  open: boolean;
+  setOpen: (value: boolean) => void;
+  type: 'sharehouse' | 'category';
+}
+
+type FormSchema = TShareHouseNameSchema | TCategoryNameSchema;
+
+const getSchema = (type: 'sharehouse' | 'category') => {
+  return type === 'sharehouse'
+    ? shareHouseNameSchema //If type is 'sharehouse
+    : categoryNameSchema; //If type is 'category'
+};
+
+/**
+ * A drawer component to edit a sharehouse or category name
+ *
+ * @param name - The current name to be edited
+ * @param open - The state of the drawer
+ * @param setOpen - The function to set the state of the drawer
+ * @param type - The type to determine if editing a 'sharehouse' or 'category'
+ *
+ * @example
+ * const [open, setOpen] = useState(false);
```

Comment:
Thank you Rina!
It is a tiny fix suggestion but could you consider use same constant 
`isDrawerOpen: open`,  `setIsDrawerOpen: setOpen` for TSDoc?